### ISSUE TITLE
Cached bundler, rubygems and bundler (1.16.1) fail together

### ISSUE THREAD

**Issue by PragTob** [travis-ci/travis-ci#9062]
This is basically https://github.com/rubygems/rubygems/issues/2055 - it seems to start failing with bundler 1.16.1

I went through a couple of iterations trying to fix this in  simplecov here: https://github.com/colszowka/simplecov/pull/648 (has builds at different stages, see commit messages).

It seems that clearing the cache from travis (removing `cache: bundler`) fixes the problem. I'm still not sure where the problem lies - probably a change in bundler 1.16.1 triggered it but it might just be exposing a problem elsewhere. I.e. in the rubygems thread it says that rubygems 2.7.3 seems to have fixed the problem, which seems untrue as me and others had failures way afterwards (with 2.7.4).

Opening this so that someone hopefully can have a look at how travis caching fits into all of this.

Thanks for a great CI free for OSS :)

**Comment by BanzaiMan** [comment]
Is the build failing without the cache? If so, please include a relevant log URL.

**Comment by PragTob** [comment]
For instance:

* failing with cache: https://travis-ci.org/colszowka/simplecov/jobs/328437403
* passing without cache: https://travis-ci.org/colszowka/simplecov/jobs/328446037

In the build overview you see some ruby versions passing, that's only because I drive by upgraded the ruby versions (which invalidated the caches and therefore also made them work).

**Comment by ms-ati** [comment]
👍  to @PragTob for a nice job explaining this issue upstream -- hopefully your discovery that removing the cache fixes the issue will now inform a broader fix for all Travis users! :)

**Comment by y-yagi** [comment]
I encountered the same issue. I tried reproducing the issue with the simple application.  

* Enable cache: https://travis-ci.org/y-yagi/reproduce-travis-error/jobs/334036008 
* Disable cache: https://travis-ci.org/y-yagi/reproduce-travis-error/jobs/334035812

Repository is [here](https://github.com/y-yagi/reproduce-travis-error/pull/3).

I tried build with several patterns. The results are as follows. 

* Problems occur when cache bundler with Linux + Ruby 2.5. It does not occur with Ruby 2.4.  
* Problems occur when updating bundler. If do not update bundler and leave it as default (bundler 1.16.0), the problem does not occur. 

Thanks.
